sdk/src/utils/scroll.rs

```diff
@@ -0,0 +1,132 @@
+use dioxus::prelude::*;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// Scroll metrics.
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct ScrollMetrics {
+    /// Current scroll position from top: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTop
+    pub scroll_top: f64,
+    /// Current scroll position from left: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollLeft
+    pub scroll_left: f64,
+
+    /// Viewport height: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight
+    pub client_height: f64,
+    /// Viewport width: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollWidth
+    pub client_width: f64,
+
+    /// Content height: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight
+    pub scroll_height: f64,
+    /// Content width: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollWidth
+    pub scroll_width: f64,
+}
+
+// Static counter to generate unique IDs for each scroll tracker instance
+static SCROLL_TRACKER_COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+/// Creates a signal that tracks root scrolling.
+/// ```rust
+/// use dioxus::{logger::tracing::{info, Level}, prelude::*};
+/// use dioxus_sdk::utils::scroll::use_root_scroll;
+///
+/// fn main() {
+///     dioxus::logger::init(Level::TRACE).unwrap();
+///     launch(App);
+/// }
+///
+/// #[component]
+/// fn App() -> Element {
+///     let random_text = "This is some random repeating text. ".repeat(1000);
+///     
+///     let scroll_metrics = use_root_scroll();
+///     use_effect(move || {
+///         let scroll_metrics = scroll_metrics();
+///         let distance_from_bottom = scroll_metrics.scroll_height - (scroll_metrics.scroll_top + scroll_metrics.client_height);
+///         info!("Distance from bottom: {}", distance_from_bottom);
+///         let scroll_percentage = (scroll_metrics.scroll_top + scroll_metrics.client_height) / scroll_metrics.scroll_height;
+///         info!("Scroll percentage: {}", scroll_percentage);
+///     });
+///  
+///     rsx! {
+///         p { "{random_text}" }  
+///     }
+/// }
+/// ```
+pub fn use_root_scroll() -> Signal<ScrollMetrics> {
+    let callback_name = use_hook(|| {
+        let instance_id = SCROLL_TRACKER_COUNTER.fetch_add(1, Ordering::SeqCst);
+        format!("scrollCallback_{}", instance_id)
+    });
+
+    let mut scroll_metrics = use_signal(|| ScrollMetrics {
+        scroll_top: 0.0,
+        scroll_left: 0.0,
+        client_height: 0.0,
+        client_width: 0.0,
+        scroll_height: 0.0,
+        scroll_width: 0.0,
+    });
+
+    let future_callback_name = callback_name.clone();
+    use_future(move || {
+        let future_callback_name = future_callback_name.clone();
+        async move {
+            let callback_name = future_callback_name;
```

Comment:
Just a suggestion to simplify this part:
```suggestion
    use_future({
        to_owned![callback_name];
        move || {
            to_owned![callback_name];
            async move {
```

Reply:
That would move `callback_name` into the future, while it is needed in the `use_drop` callback https://github.com/DioxusLabs/sdk/pull/75/files#diff-698efcc4a2d34cdd5d48f5333cda6739c2d082d86ae9ee981b83c861a57ed62aR123 , so `future_callback_name` rename is needed, so `to_owned!` can't be used there.

In the other cases, `to_owned!` seems to be accomplishing the same thing, just less explicitly, and I would then have to rename all the inners to `future_callback_name`.

Reply:
Done. PR seems to be stuck in "processing updates" for the past 10 min though.

Reply:
10 hours later and still processing.. weird, had to force push, but it is showing now.